Updated Ash dependency from 0.24 to 0.26

keringar opened the pull request:
Fixes #2510. A lot of the changes were just renaming various constants that were moved around with no major changes. I went through the diff and checked to make sure everything was updated correctly, but it's possible there could have been a typo somewhere in the 500 or so changes. 

I tested the changes with warden and the examples and nothing appears to be wrong, but I can't be sure since it looks like warden is broken with 7 failing tests on the Vulkan backend. But this appears to be a separate issue from updating Ash since warden is also broken on at least the 0.1 release and maybe earlier.

PR checklist:
- [x] `make` succeeds (on *nix)
- [ ] `make reftests` succeeds
- [x] tested examples with the following backends:
    - vulkan
- [x] `rustfmt` run on changed code

kvark commented:
The failures are fixed by #2534 , fwiw

kvark commented on src/backend/vulkan/src/conv.rs:
```diff
@@ -208,86 +208,86 @@ pub fn map_mip_filter(filter: image::Filter) -> vk::SamplerMipmapMode {
+        Primitive::LineList               => vk::PrimitiveTopology::LINE_LIST,
+        Primitive::LineListAdjacency      => vk::PrimitiveTopology::LINE_LIST_WITH_ADJACENCY,
+        Primitive::LineStrip              => vk::PrimitiveTopology::LINE_STRIP,
+        Primitive::LineStripAdjacency     => vk::PrimitiveTopology::LINE_LIST_WITH_ADJACENCY,
```
this should be `LINE_STRIP_WITH_ADJACENCY`

keringar replied:
Fixed

kvark commented on src/backend/vulkan/src/lib.rs:
```diff
@@ -395,21 +401,18 @@ impl hal::PhysicalDevice<Backend> for PhysicalDevice {
-                        ash::DeviceError::LoadError(err) => panic!("{:?}", err),
-                        ash::DeviceError::VkError(err) => Into::<result::Error>::into(err),
-                    })
+                    .map_err(|err| Into::<result::Error>::into(err))
```
nit: could just be `map_err(Into::<result::Error>::into)`

keringar replied:
Fixed

keringar commented:
I squashed and ran rustfmt. The failing Windows builds were also fixed. While I don't have access to an Android device to test, I believe the android stuff in `window.rs` has also been correctly updated.

kvark commented:
Thanks!
Bors r+

> On Dec 29, 2018, at 02:37, Warren Hu <notifications@github.com> wrote:
> 
> I squashed and ran rustfmt. The failing Windows builds were also fixed. While I don't have access to an Android device to test, I believe the android stuff in window.rs has also been correctly updated.
> 
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub, or mute the thread.
